compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingInterface.cpp

```diff
@@ -304,7 +539,15 @@ struct TileDispatchUsingSCFForOp
 
   LogicalResult matchAndRewrite(TilingInterface op,
                                 PatternRewriter &rewriter) const override {
-    return returningMatchAndRewrite(op, rewriter);
+    auto result = returningMatchAndRewrite(op, rewriter);
+    if (failed(result)) return failure();
+
+    if (clSpecializeWorkgroupDistribution) {
```

Comment:
I think this should be decided by the caller as there may be different strategy to handle partial tiles (ex: padding, etc...) that may happen later.

Reply:
BTW `clSpecializeWorkgroupDistribution` could be a pass option instead. This should work well with making it controllable by caller too

Reply:
Thanks for the suggestion! Let me take a look.

Reply:
Done!